pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go

```diff
@@ -275,151 +237,28 @@ func (w *Watcher) handlePluginRegistration(socketPath string) error {
 		return fmt.Errorf("failed to get plugin info using RPC GetInfo at socket %s, err: %v", socketPath, err)
 	}
 
```

Comment:
We should move all the above code (`dial`, `context.WithTimeout`, and `GetInfo`) to the new Register operation as well. If we leave it here, and it fails, we will never retry. If it is part of the new Register operation, the reconciler+goroutinemap will handle retries with exponential backoff.

Reply:
@saad-ali That makes sense. That means we'll also need to update the input around `DesiredStateOfWorld`'s `AddOrUpdatePlugin` and `ActualStateOfWorld` since we won't have the full plugin info before we call `GenerateRegisterPluginFunc`